### PR TITLE
Language Server: Parse pragma arguments declarations and show codelenses

### LINKED ISSUE #431: Language server command to get entry point pragma arguments
### Issue To Be Solved

Language server clients need a mechanism to get the arguments that should be used when sending a transaction or executing a script.

### Suggest A Solution

- [x] Parse the docstring of the entrypoint for all arguments pragmas in the form `/// pragma arguments <argument-list>`, where `<argument-list>` is a Cadence argument list, e.g. `(a: 1)`
- [x] Show code lenses for entry points
- [x] Change script execution command and transaction submission commands to accept arguments

### PR DESCRIPTION
**Pull request by turbolent**
Closes #431

- Parse docstring for transactions and retain it in the AST element
- Improve tests for language server: log error of language server when it exits unsuccessfully
- Add a function to parse an argument list where all arguments are literals, e.g. `(1, true)`
- Add a function to parse all `pragma arguments` declarations in a docstring, e.g. `/// pragma arguments (1, true)`
- Refactor codelenses for transactions and scripts: consider pragma arguments declarations, show one codelens per argument list. Include the arguments in the command sent from the client to the server
- Accept arguments in the script execution command and the transaction submission command

Examples:

- Script:
  Screen Shot 2020-10-29 at 6 42 12 PM

- Tranaction:
  Screen Shot 2020-10-29 at 6 41 18 PM

### CODE REVIEW

**Review by jordanschalm — APPROVED** [review]

File: languageserver/integration/codelenses.go [hunk 1]
```diff
@@ -160,30 +135,94 @@ func (i *FlowIntegration) showDeployContractInterfaceAction(
 	}
 }
 
-func (i *FlowIntegration) showExecuteScriptAction(
+// entryPointActions shows an execute button when there is exactly one valid entry point
+// (valid script function or transaction declaration) and no other actionable declarations.
+//
+// NOTE: requires
```
Comment (turbolent) [inline languageserver/integration/codelenses.go:141]:
```suggestion
```

**Comment by turbolent** [comment]
This PR can't be merged as is at the moment: It contains changes to both Cadence, and to the language server. However, the language server is its own module and depends on Cadence. Therefore the changes to Cadence must be merged and released first, then the language server must be updated to use the new functionality in Cadence.

**Comment by turbolent** [comment]
Opened a PR for the first part: #528

**Comment by turbolent** [comment]
#528 is merged, I updated this PR to use the new functionality introduced in v0.12.4

**Comment by psiemens** [comment]
🎉

**Pull request merged**